compiler/src/dotty/tools/dotc/core/tasty/TreeUnpickler.scala

```diff
@@ -278,7 +278,8 @@ class TreeUnpickler(reader: TastyReader, tastyName: TastyName.Table, posUnpickle
               result
             case METHODtype =>
               val (names, paramReader) = readNamesSkipParams
-              val result = MethodType(names.map(_.toTermName), paramReader.readParamTypes[Type](end))(
+              val result = MethodType(names.map(_.toTermName))(
+                mt => paramReader.readParamTypes[Type](end), // !!!
```

Comment:
That "!!!" is slightly alarming :-)

I suppose you are referring to the fact that we now rely on the constructor calling the two closures in the right order, which is somewhat risky indeed. Why not read them explicitly and construct a constant function?